compiler/rustc_codegen_ssa/src/base.rs

```diff
@@ -622,31 +624,33 @@ pub fn codegen_crate<B: ExtraBackendMethods>(
     // This likely is a temporary measure. Once we don't have to support the
     // non-parallel compiler anymore, we can compile CGUs end-to-end in
     // parallel and get rid of the complicated scheduling logic.
-    let pre_compile_cgus = |cgu_reuse: &[CguReuse]| {
-        if cfg!(parallel_compiler) {
-            tcx.sess.time("compile_first_CGU_batch", || {
-                // Try to find one CGU to compile per thread.
-                let cgus: Vec<_> = cgu_reuse
-                    .iter()
-                    .enumerate()
-                    .filter(|&(_, reuse)| reuse == &CguReuse::No)
-                    .take(tcx.sess.threads())
-                    .collect();
-
-                // Compile the found CGUs in parallel.
-                let start_time = Instant::now();
-
-                let pre_compiled_cgus = par_iter(cgus)
-                    .map(|(i, _)| {
-                        let module = backend.compile_codegen_unit(tcx, codegen_units[i].name());
-                        (i, module)
-                    })
-                    .collect();
-
-                (pre_compiled_cgus, start_time.elapsed())
-            })
-        } else {
-            (FxHashMap::default(), Duration::new(0, 0))
+    let pre_compile_cgus = |_cgu_reuse: &[CguReuse]| {
+        cfg_if::cfg_if! {
+            if #[cfg(parallel_compiler)] {
+                tcx.sess.time("compile_first_CGU_batch", || {
+                    // Try to find one CGU to compile per thread.
+                    let cgus: Vec<_> = _cgu_reuse
+                        .iter()
+                        .enumerate()
+                        .filter(|&(_, reuse)| reuse == &CguReuse::No)
+                        .take(tcx.sess.threads())
+                        .collect();
+
+                    // Compile the found CGUs in parallel.
+                    let start_time = Instant::now();
+
+                    let pre_compiled_cgus = par_iter(cgus)
+                        .map(|(i, _)| {
+                            let module = backend.compile_codegen_unit(tcx, codegen_units[i].name());
+                            (i, module)
+                        })
+                        .collect();
+
+                    (pre_compiled_cgus, start_time.elapsed())
+                })
+            } else {
+                (FxHashMap::default(), Duration::new(0, 0))
+            }
```

Comment:
Could you make two closures? This would allow to avoid the `cfg-if` dependency:
```
#[cfg(parallel_compiler)]
let pre_compile_cgus = |cgu_reuse: &[CguReuse]| { /* ... */ };

#[cfg(not(parallel_compiler))]
let pre_compile_cgus = |_: &[CguReuse]| (FxHashMap::default(), Duration::new(0, 0));
```

Reply:
Done.